internal/pkg/deploy/cloudformation/task.go

```diff
@@ -48,3 +50,67 @@ func (cf CloudFormation) DeployTask(input *deploy.CreateTaskResourcesInput, opts
 
 	return nil
 }
+
+// ListTaskStacks returns all the CF stacks which represent one-off copilot tasks in a given application's environments
+func (cf CloudFormation) ListTaskStacks(appName, envName string) ([]deploy.TaskStackInfo, error) {
+	taskAppEnvTags := map[string]string{
+		deploy.TaskTagKey: "",
+		deploy.AppTagKey:  appName,
+		deploy.EnvTagKey:  envName,
+	}
+	tasks, err := cf.cfnClient.ListStacksWithTags(taskAppEnvTags)
+
+	if err != nil {
+		return nil, err
+	}
+	var outputTaskStacks []deploy.TaskStackInfo
+	for _, task := range tasks {
+
+		outputTaskStacks = append(outputTaskStacks, deploy.TaskStackInfo{
+			StackName: aws.StringValue(task.StackName),
+			App:       appName,
+			Env:       envName,
+
+			RoleARN: aws.StringValue(task.RoleARN),
+		})
+	}
+	return outputTaskStacks, nil
+}
+
+// ListDefaultTaskStacks returns all the CF stacks created by copilot but not associated with an application.
+func (cf CloudFormation) ListDefaultTaskStacks() ([]deploy.TaskStackInfo, error) {
+	tasks, err := cf.cfnClient.ListStacksWithTags(map[string]string{deploy.TaskTagKey: ""})
```

Comment:
I don't know if it is an overkill but looks like we might need `ListStacksWithTagFilter(wantedTags map[string]string, ignoredTags map[string]string)`. Just a thought though.

Reply:
I think we could enable this by converting the wantedTags to an option function (like `WithRequiredTags()`) then add another `WithIgnoredTags()` function if it's needed. That way the behavior can be confined to the code which generates the matcher. 

I don't think it's necessary right now though.